nixos/modules/services/web-apps/pingvin-share.md

```diff
@@ -0,0 +1,86 @@
+# Pingvin Share {#module-services-pingvin-share}
+
+A self-hosted file sharing platform and an alternative for WeTransfer.
+
+## Configuration {#module-services-pingvin-share-basic-usage}
+
+By default, the module will execute Pingvin Share backend and frontend on the ports 8080 and 3000.
+
+I will run two systemd services named `pingvin-share-backend` and `pingvin-share-frontend` in the data directory `/var/lib/pingvin-share`.
+
+Here is a basic configuration:
+
+```nix
+{ ... }:
+
+{
+  services-pingvin-share = {
+    enable = true;
+
+    openFirewall = true;
+
+    settings.dataDir = "/var/lib/data/share";
+
+    backend.port = 9010;
+    frontend.port = 9011;
+  };
+}
+```
+
+## Reverse proxy configuration {#module-services-pingvin-share-reverse-proxy-configuration}
+
+If you don't want to expose an open port (for example on a headless server), you can configure Nginx such like this:
+
+```nix
+{ ... }:
+
+{
+  services-pingvin-share = {
+    enable = true;
+
+    settings = {
+      hostname = "pingvin-share.domain.tld";
+      https = true;
+    };
+
+    nginx.enable = true;
+  };
+}
+```
+
+which is equivalent to:
```

Comment:
This section is unnecessary and will probably become outdated over time. If people are interested in the details, they can read the code. If they want to extend it, the module system supports that.